icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -196,6 +252,23 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         if (bestSnlp[numCodePts] == kint32max) {
             t_boundary[numBreaks] = numCodePts;
             numBreaks++;
+        } else if (isPhraseBreaking) {
+            t_boundary[numBreaks] = numCodePts;
+            numBreaks++;
+            int prevIdx = numCodePts;
+            for (int i = prev[numCodePts]; i > 0; i = prev[i]) {
+                int length = prevIdx - i;
+                prevIdx = i;
+                if (isPhraseBreaking) {
```

Comment:
you already have isPhraseBreaking in the else if so this if (isPhraseBreaking) is always true.
and once we remove that check, the remaining logic is simpler so how about just flip the if by adding ! in front of contains check and place the two line into the if block as below
```
            for (int i = prev[numCodePts]; i > 0; i = prev[i]) {
                int length = prevIdx - i;
                prevIdx = i;
                // Skip the breakpoint if it is after the particle or Hiragana.
                if (!fSkipSet.contains(getPatternFromText(text, s, i, length))) {
                     t_boundary[numBreaks] = i;
                     numBreaks++;
                }
            }
```
is it "after" ? or before? I am not sure. Please correct if I made mistake

Reply:
You are right. The check here should be removed.